src/Halogen/Aff/Driver/Eval.purs

```diff
@@ -37,6 +39,25 @@ type LifecycleHandlers eff =
   , finalizers :: List (Aff (HalogenEffects eff) Unit)
   }
 
+parSequenceAff_
+  :: forall eff a
+   . L.List (Aff (avar :: AV.AVAR, ref :: REF | eff) a)
+  -> Aff (avar :: AV.AVAR, ref :: REF | eff) Unit
+parSequenceAff_ as = do
+  let n = L.length as
+  when (n > 0) do
+    var <- AV.makeVar
+    ref <- liftEff $ newRef n
```

Comment:
Since we are specialized to Lists, we can get away without the `length`, and just put the actual List in the Ref, and pattern matching on the spine.